cleo/io/outputs/stream_output.py

```diff
@@ -31,7 +31,7 @@ def __init__(
         formatter: Formatter | None = None,
     ) -> None:
         self._stream = stream
-        self._supports_utf8: bool | None = None
+        self._supports_utf8: bool = True
```

Comment:
i think this breaks the utf8 detection logic, https://github.com/python-poetry/cleo/blob/06ba9ea40488fad28446b764e4c9ea3a29cb4982/cleo/io/outputs/stream_output.py#L49

maybe we can just call the detection logic immediately here (assuming it's not an expensive call)